### PR TITLE
Internationalization

### PR DESCRIPTION
**Pull request by Dougley**
This pull request adds back the internationalization support from v6 but with extra features thanks to proper support for ICU syntax.

Please review this thoroughly, I might have missed a few keys.

### CODE REVIEW

File: src/events/messageCreate/50_command-engine.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const commands = require('../../components/commands')
 const { commands: cmdAnalytics } = require('../../components/analytics')
+const i18n = require('../../internal/i18n')
```
Comment (linuswillner) [inline src/events/messageCreate/50_command-engine.js:3]: Ironically, this seems to contradict the contribution guideline that states that the `i18n` object is globally available and shouldn't be required on its own... Is there a reason for this to be here?

File: src/internal/i18n.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const dirreq = require('./dir-require')
+const logger = require('./logger')
+const languages = dirreq('src/languages/**/*.json')
+const IntlMessageFormat = require('intl-messageformat').default
+const defaultlang = process.env.WILDBEAST_LANGUAGE || 'en-EN'
```
Comment (linuswillner) [inline src/internal/i18n.js:5]: Camelcase
```suggestion
const defaultLang = process.env.WILDBEAST_LANGUAGE || 'en-EN'
```

File: src/internal/i18n.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+logger.debug('I18N', `Loaded ${Object.keys(languages).length} language(s), using ${defaultlang} as default`)
```
Comment (linuswillner) [inline src/internal/i18n.js:7]:
```suggestion
logger.debug('I18N', `Loaded ${Object.keys(languages).length} language(s), using ${defaultLang} as default`)
```

File: src/internal/i18n.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+module.exports = {
+  t (key, ctx, lang) {
+    try {
+      const msg = key.split('.').reduce((o, i) => o[i], languages[lang] || languages[defaultlang])
```
Comment (linuswillner) [inline src/internal/i18n.js:12]:
```suggestion
      const msg = key.split('.').reduce((o, i) => o[i], languages[lang] || languages[defaultLang])
```

File: src/internal/i18n.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      return new IntlMessageFormat(msg, defaultlang).format(ctx)
```
Comment (linuswillner) [inline src/internal/i18n.js:13]:
```suggestion
      return new IntlMessageFormat(msg, defaultLang).format(ctx)
```

**Review by linuswillner — CHANGES_REQUESTED** [review]
Some minor nits here and there. Also, standardisation for the punctuation in translation strings is needed as discussed privately.

File: src/events/messageCreate/50_command-engine.js (hunk 1 above)
Comment (Dougley) [reply src/events/messageCreate/50_command-engine.js:3]: VSCode auto-import 💯

**Review by curtisf — APPROVED** [review]
not very good with grammar so I just briefly looked at i18n english strings and didn't see anything wrong

**Review by linuswillner — APPROVED** [review]

**Review by vDelite — CHANGES_REQUESTED** [review]
"noResults": "Nothing found with your search query"
I'm not a fan of this string, I think something like "No search results found" would be better

    "disclaimer": "Please note: translations are provided by the community, we cannot guarantee their correctness, completeness, or quality",
Add an "and" after the first comma

      "siteNotSupported": "I don't have support for {site}, currently I support {supported}"
"I don't currently support {site}. Only {supported} are supported." instead?

**Comment by Dougley** [comment]
> "I don't currently support {site}. Only {supported} are supported." instead?

This looks weird on Discord due to "and" missing

afbeelding

**Review by vDelite — APPROVED** [review]

**Pull request merged**